File: 1909-buildings-with-an-ocean-view/buildings-with-an-ocean-view.py
class Solution:
    def findBuildings(self, heights: List[int]) -> List[int]:
        
        stack = []
        ans = []
        for i,h in enumerate(heights):
            while stack and stack[-1] <= h:
                stack.pop()
                ans.pop()
            stack.append(h)
            ans.append(i)

        return ans

File: 1909-buildings-with-an-ocean-view/buildings-with-an-ocean-view.cpp
class Solution {
public:
    vector<int> findBuildings(vector<int>& heights) {
        int maxh = 0;
        int n = heights.size();
        vector<int> arr;

        for (int i = n-1; i >= 0; i--){
            if (heights[i] > maxh){
                maxh = heights[i];
                arr.push_back(i);
            }
        }

        reverse(arr.begin(),arr.end());
        return arr;
        
    }
};